fem/nonlinearform_ext.hpp

```diff
@@ -33,15 +37,21 @@ class NonlinearFormExtension : public Operator
 
    /// Assemble at the AssemblyLevel of the subclass.
    virtual void Assemble() = 0;
-   /// Assemble gradient data at the AssemblyLevel of the subclass, for the
-   /// state @a x, which is assumed to be a ldof Vector.
-   virtual void AssembleGradient(const Vector &x) = 0;
 
-   /// Assumes that @a x is a ldof Vector.
+   /** @brief Return the gradient as an L-to-L Operator. The input @a x must be
+       an L-vector (i.e. GridFunction-size vector). */
+   /** Essential boundary conditions are NOT applied to the returned operator.
+
+       The returned gradient Operator defines the virtual method GetProlongation
+       which enables support for the method FormSystemOperator to define the
+       matrix-free global true-dof gradient with imposed boundary conditions. */
    virtual Operator &GetGradient(const Vector &x) const = 0;
 
    /// Assumes that @a x is a ldof Vector.
```

Comment:
Should we change `ldof Vector` to `L-vector` in all comments?

Reply:
I don't feel it is necessary but if you want, we can do it.